### PR TITLE
Support Opencv4.2(load opencv dependency)

### PR DESCRIPTION
**Pull request by Le-Zheng**
Compile OpenCV4.2:
`cmake -D CMAKE_BUILD_TYPE=RELEASE -D INSTALL_PYTHON_EXAMPLES=OFF -D INSTALL_C_EXAMPLES=OFF -D BUILD_DOCS=OFF -D BUILD_opencv_python2=OFF -D BUILD_opencv_python3=OFF -DWITH_1394=OFF -DBUILD_TESTS=OFF -DBUILD_EXAMPLES=OFF -DBUILD_SHARED_LIBS=ON -DBUILD_JASPER=ON -DBUILD_JPEG=ON -DBUILD_PNG=ON -DBUILD_OPENCL=OFF -DOPENCV_DNN_OPENCL=OFF -DWITH_PNG=ON -DWITH_MKL=OFF -DMKL_WITH_TBB=OFF -DMKL_USE_MULTITHREAD=OFF -DMKL_WITH_OPENMP=OFF -DWITH_ITT=OFF -DWITH_FFMPEG=OFF -DWITH_GSTREAMER=OFF -DWITH_DSHOW=OFF -DWITH_MSMF=OFF -DBUILD_PERF_TESTS=OFF -DBUILD_IPP_IW=OFF -DBUILD_ITT=OFF -DBUILD_opencv_apps=OFF -DBUILD_opencv_calib3d=OFF -DBUILD_opencv_dnn=OFF -DBUILD_opencv_features2d=OFF -DBUILD_opencv_flann=OFF -DBUILD_opencv_gapi=OFF -DBUILD_opencv_highgui=OFF -DBUILD_opencv_imgcodecs=ON -DBUILD_opencv_imgproc=ON -DBUILD_opencv_java=ON -DBUILD_opencv_java_bindings_generator=ON -DBUILD_opencv_js=OFF -DBUILD_opencv_ml=OFF -DBUILD_opencv_objdetect=OFF -DBUILD_opencv_photo=OFF -DBUILD_opencv_python2=OFF -DBUILD_opencv_python3=OFF -DBUILD_opencv_python_bindings_generator=OFF -DBUILD_opencv_python_tests=OFF -DBUILD_opencv_stitching=OFF -DBUILD_opencv_video=OFF -DBUILD_opencv_videoio=OFF -DBUILD_PROTOBUF=OFF -DWITH_IPP=OFF -DWITH_GTK=OFF -DWITH_OPENEXR=OFF -DWITH_JASPER=OFF -DWITH_WEBP=OFF -DWITH_OPENCL=OFF -DWITH_V4L=OFF -DWITH_VTK=OFF ..
`

### CODE REVIEW

**Comment by Le-Zheng** [comment]
local test with BigDL Dllib UTs passed.

File: README.md [hunk 1]
```diff
@@ -31,9 +31,9 @@ XCode 9.0
 
 ## Building from Source
 
-### Prepare OpenCV 3.2.0
+### Prepare OpenCV 4.2.0
 
-Download `opencv-320.jar` from maven. Copy this jar to `/opt/opencv` dir. Extract `libopencv_java320.so` (Linux) or `libopencv_java320.dylib` (macOS) from `opencv-320.jar`, and move to `/opt/opencv`.
+Download `opencv-420.jar` from maven. Copy this jar to `/opt/opencv` dir. Extract `libopencv_java420.so` (Linux) or `libopencv_java420.dylib` (macOS) from `opencv-420.jar`, and move to `/opt/opencv`.
 
 Or you can build this 2 files from OpenCV source code.
```
Comment (qiyuangong) [inline README.md:39]: Need to update cmake command in readme.

File: README.md [hunk 2]
```diff
@@ -51,9 +51,9 @@ cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -DBUILD_TES
 make && make install
 # Copy jar & libs to /opt/opencv
 mkdir /opt/opencv
-scp ./bin/opencv-420.jar /opt/opencv
```
Comment (qiyuangong) [inline README.md:54]: Maybe we can move this doc to opencv dir. Then, add link to the readme

**Review by qiyuangong — APPROVED** [review]
LGTM

**Pull request merged**